<!doctype html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">

<!-- Begin Jekyll SEO tag v2.4.0 -->
<title>Transcriptomic protocols | NGS protocols to wrangling Transcriptomics.</title>
<meta name="generator" content="Jekyll v3.7.3" />
<meta property="og:title" content="Transcriptomic protocols" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="NGS protocols to wrangling Transcriptomics." />
<meta property="og:description" content="NGS protocols to wrangling Transcriptomics." />
<link rel="canonical" href="http://localhost:4000/markdown/transrate.html" />
<meta property="og:url" content="http://localhost:4000/markdown/transrate.html" />
<meta property="og:site_name" content="Transcriptomic protocols" />
<script type="application/ld+json">
{"description":"NGS protocols to wrangling Transcriptomics.","@type":"WebPage","url":"http://localhost:4000/markdown/transrate.html","headline":"Transcriptomic protocols","@context":"http://schema.org"}</script>
<!-- End Jekyll SEO tag -->


    <link rel="stylesheet" href="/assets/css/style.css?v=">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <a href=""> <h1>Transcriptomic protocols</h1> </a>
        <p>NGS protocols to wrangling Transcriptomics.</p>

        

        

        
          <ul>
            <li><a href="">Download <strong>ZIP File</strong></a></li>
            <li><a href="">Download <strong>TAR Ball</strong></a></li>
            <li><a href="">View On <strong>GitHub</strong></a></li>
          </ul>
        
      </header>
      <section>

      <p>La idea es cuantificar el rango de tipo de errores conocidos que ocurren durante el ensamble denovo de transcriptomas:</p>

<p>Error type:</p>
<ul>
  <li>Family collapse (gene AA, AB, and AC collapsed as a single hybrid contig…bases in read as a disagrement in their entropy)</li>
  <li>Chimeris (contig compuse from two disting transcripts (may or may not be related)…coverage is non uniform)</li>
  <li>Unsupported insertion (the new contig contain a fragment inserted along them, therefore pseudo-contig have not good aligned…no reads align to insertion)</li>
  <li>Incompletness (the contig lengh is small in comparison with the reads-gene where it came…read pairs align off end of contig)</li>
  <li>Fragmentation (A transcript result in a small diferent set of contig …result in bridging reads pairs)</li>
  <li>Local misassembly (transcript assembly is in a wrong orientation afer k-mering…read pairs in wrong orientation …These are detectable when both members of a read pair align to a single contig, but in a manner inconsistent with the sequencing protocol) &lt;– no detected in a non-strand mode setup</li>
  <li>Redundancy (A transcript result in a diferent set of contigs … all reads assign to best contig)</li>
</ul>

<p><a href="https://genome.cshlp.org/content/early/2016/06/01/gr.196469.115.full.pdf+html">transrate paper</a></p>

<p><img src="../figures/transrate.png" alt="" /></p>

<p>Whinin your <em>work directory</em> lets make a new directory with name <em>transrate</em> and go in to start working in:</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">mkdir </span>transrate <span class="o">&amp;&amp;</span> <span class="nb">cd </span>transrate 
</code></pre></div></div>

<p>Lets to use the command shell <code class="highlighter-rouge">ln -s</code> in order to make a symbolic link from the assembly with name Trinity.fasta:</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">ln</span> <span class="nt">-s</span> ../trinity_out/Trinity.fasta <span class="nb">.</span>
</code></pre></div></div>

<p>Then, concatenate in a unique dataset the left and right libraries for input in transrate; both data sets are necessary for transrating:</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>zcat ../trinity_out/<span class="k">*</span>R1<span class="k">*</span>P.qtrim.gz <span class="o">&gt;</span> R1.P.qtrim.fq &amp;
zcat ../trinity_out/<span class="k">*</span>R2<span class="k">*</span>P.qtrim.gz <span class="o">&gt;</span> R2.P.qtrim.fq 
</code></pre></div></div>

<p>Now export the source from the transrate tool; and finally run transrate (considere to use the sbatch task manager in the cluster)</p>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">tool</span><span class="o">=</span>/LUSTRE/bioinformatica_data/RNA/ricardo/bioinformatics/transrate-1.0.3-linux-x86_64/
<span class="nb">export </span><span class="nv">PATH</span><span class="o">=</span><span class="nv">$tool</span>:<span class="nv">$PATH</span>
</code></pre></div></div>

<div class="language-shell highlighter-rouge"><div class="highlight"><pre class="highlight"><code>transrate <span class="se">\</span>
    <span class="nt">--assembly</span> Trinity.fasta <span class="se">\</span>
    <span class="nt">--left</span> R1.P.qtrim.fq <span class="se">\</span>
    <span class="nt">--right</span> R2.P.qtrim.fq <span class="se">\</span>
    <span class="nt">--threads</span> 96 <span class="se">\</span>
    <span class="nt">--output</span> gcontigs 1&gt; transrate.log &amp;
</code></pre></div></div>

<p>The <code class="highlighter-rouge">transrate.log</code> file include some statistics like:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[ INFO] 2018-03-02 00:41:37 : Calculating contig metrics...
[ INFO] 2018-03-02 00:41:56 : Contig metrics:
[ INFO] 2018-03-02 00:41:56 : -----------------------------------
[ INFO] 2018-03-02 00:41:56 : n seqs                       147454
[ INFO] 2018-03-02 00:41:56 : smallest                        201
[ INFO] 2018-03-02 00:41:56 : largest                       10795
[ INFO] 2018-03-02 00:41:56 : n bases                   108715008
[ INFO] 2018-03-02 00:41:56 : mean len                     737.28
[ INFO] 2018-03-02 00:41:56 : n under 200                       0
[ INFO] 2018-03-02 00:41:56 : n over 1k                     33509
[ INFO] 2018-03-02 00:41:56 : n over 10k                        1
[ INFO] 2018-03-02 00:41:56 : n with orf                    40996
[ INFO] 2018-03-02 00:41:56 : mean orf percent              58.76
[ INFO] 2018-03-02 00:41:56 : n90                             308
[ INFO] 2018-03-02 00:41:56 : n70                             640
[ INFO] 2018-03-02 00:41:56 : n50                            1121
[ INFO] 2018-03-02 00:41:56 : n30                            1754
[ INFO] 2018-03-02 00:41:56 : n10                            2818
[ INFO] 2018-03-02 00:41:56 : gc                             0.39
[ INFO] 2018-03-02 00:41:56 : bases n                           0
[ INFO] 2018-03-02 00:41:56 : proportion n                    0.0
[ INFO] 2018-03-02 00:41:56 : Contig metrics done in 19 seconds
[ INFO] 2018-03-02 00:41:56 : Calculating read diagnostics...
[ INFO] 2018-03-02 01:00:20 : Read mapping metrics:
[ INFO] 2018-03-02 01:00:20 : -----------------------------------
[ INFO] 2018-03-02 01:00:20 : fragments                  51394897
[ INFO] 2018-03-02 01:00:20 : fragments mapped           19860500
[ INFO] 2018-03-02 01:00:20 : p fragments mapped             0.39
[ INFO] 2018-03-02 01:00:20 : good mappings              17246823
[ INFO] 2018-03-02 01:00:20 : p good mapping                 0.34
[ INFO] 2018-03-02 01:00:20 : bad mappings                2613677
[ INFO] 2018-03-02 01:00:20 : potential bridges                 0
[ INFO] 2018-03-02 01:00:20 : bases uncovered            68043780
[ INFO] 2018-03-02 01:00:20 : p bases uncovered              0.63
[ INFO] 2018-03-02 01:00:20 : contigs uncovbase            124374
[ INFO] 2018-03-02 01:00:20 : p contigs uncovbase            0.84
[ INFO] 2018-03-02 01:00:20 : contigs uncovered            147454
[ INFO] 2018-03-02 01:00:20 : p contigs uncovered             1.0
[ INFO] 2018-03-02 01:00:20 : contigs lowcovered           147454
[ INFO] 2018-03-02 01:00:20 : p contigs lowcovered            1.0
[ INFO] 2018-03-02 01:00:20 : contigs segmented             15457
[ INFO] 2018-03-02 01:00:20 : p contigs segmented             0.1
[ INFO] 2018-03-02 01:00:20 : Read metrics done in 1103 seconds
[ INFO] 2018-03-02 01:00:20 : No reference provided, skipping comparative diagnostics
[ INFO] 2018-03-02 01:00:40 : TRANSRATE ASSEMBLY SCORE     0.0297
[ INFO] 2018-03-02 01:00:40 : -----------------------------------
[ INFO] 2018-03-02 01:00:40 : TRANSRATE OPTIMAL SCORE       0.123
[ INFO] 2018-03-02 01:00:40 : TRANSRATE OPTIMAL CUTOFF     0.2353
[ INFO] 2018-03-02 01:00:41 : good contigs                  65516
[ INFO] 2018-03-02 01:00:41 : p good contigs                 0.44
[ INFO] 2018-03-02 01:00:41 : Writing contig metrics for each contig to /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/oyster-rawdata/gonada_all_sex_vs_hc/transrate/gcontigs/Trinity/contigs.csv
[ INFO] 2018-03-02 01:01:02 : Writing analysis results to assemblies.csv

</code></pre></div></div>
<p>The optimal cutoff from 0.2352 were calculated with transrate and used to keep good contigs assembled (from the total contigs 44% were considere as good).  Try comparing results with the <a href="denovo-Assembly">previous Assembly quality statistics</a> which came from the same set of libraries. Are there some difference ?</p>

<p>lets start into R</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">contigs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">".gcontigs/Trinity/contigs.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="nb">T</span><span class="p">)</span><span class="w">

</span><span class="n">contigs</span><span class="p">[</span><span class="s1">'group'</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NA</span><span class="w">
</span><span class="n">contigs</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">contigs</span><span class="p">[</span><span class="s1">'score'</span><span class="p">]</span><span class="w"> </span><span class="o">==</span><span class="w">  </span><span class="m">0.0297</span><span class="p">),</span><span class="s1">'group'</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s1">'Assembly Score'</span><span class="w">
</span><span class="n">contigs</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">contigs</span><span class="p">[</span><span class="s1">'score'</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;=</span><span class="w">  </span><span class="m">0.123</span><span class="p">),</span><span class="s1">'group'</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s1">'Optimal Score'</span><span class="w">
</span><span class="n">contigs</span><span class="p">[</span><span class="n">which</span><span class="p">(</span><span class="n">contigs</span><span class="p">[</span><span class="s1">'score'</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;=</span><span class="m">0.2353</span><span class="p">),</span><span class="s1">'group'</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s1">'Optimal Cutoff'</span><span class="w">

</span><span class="n">table</span><span class="p">(</span><span class="n">contigs</span><span class="o">$</span><span class="n">group</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h6 id="header-6"><a href="#header-6"></a>Header 6</h6>

<table>
  <thead>
    <tr>
      <th style="text-align: left">head1</th>
      <th style="text-align: left">head two</th>
      <th style="text-align: left">three</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">ok</td>
      <td style="text-align: left">good swedish fish</td>
      <td style="text-align: left">nice</td>
    </tr>
    <tr>
      <td style="text-align: left">out of stock</td>
      <td style="text-align: left">good and plenty</td>
      <td style="text-align: left">nice</td>
    </tr>
    <tr>
      <td style="text-align: left">ok</td>
      <td style="text-align: left">good <code class="highlighter-rouge">oreos</code></td>
      <td style="text-align: left">hmm</td>
    </tr>
    <tr>
      <td style="text-align: left">ok</td>
      <td style="text-align: left">good <code class="highlighter-rouge">zoute</code> drop</td>
      <td style="text-align: left">yumm</td>
    </tr>
  </tbody>
</table>

<h3 id="theres-a-horizontal-rule-below-this">There’s a horizontal rule below this.</h3>

<hr />

<h3 id="here-is-an-unordered-list">Here is an unordered list:</h3>

<ul>
  <li>Item foo</li>
  <li>Item bar</li>
  <li>Item baz</li>
  <li>Item zip</li>
</ul>

<h3 id="and-an-ordered-list">And an ordered list:</h3>

<ol>
  <li>Item one</li>
  <li>Item two</li>
  <li>Item three</li>
  <li>Item four</li>
</ol>

<h3 id="and-a-nested-list">And a nested list:</h3>

<ul>
  <li>level 1 item
    <ul>
      <li>level 2 item</li>
      <li>level 2 item
        <ul>
          <li>level 3 item</li>
          <li>level 3 item</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<h3 id="definition-lists-can-be-used-with-html-syntax">Definition lists can be used with HTML syntax.</h3>

<dl>
<dt>Name</dt>
<dd>Godzilla</dd>
<dt>Born</dt>
<dd>1952</dd>
<dt>Birthplace</dt>
<dd>Japan</dd>
<dt>Color</dt>
<dd>Green</dd>
</dl>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Long, single-line code blocks should not wrap. They should horizontally scroll if they are too long. This line should be long enough to demonstrate this.
</code></pre></div></div>

<p>Text can be <strong>bold</strong>, <em>italic</em>, or <del>strikethrough</del>.</p>

<p><a href="../">back to Menu</a></p>

<p><a href="another-page">Link to another page</a>.</p>


      </section>
      <footer>
        
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="/assets/js/scale.fix.js"></script>


  
  </body>
</html>
